docs/source/private-data/private-data.md

```diff
@@ -140,6 +143,184 @@ documentation on [transaction flow](../txflow.html).
+  data on their node or the regulator's node.
+
+* Buyer verifies a hash of the private details matches the on-chain public hash.
+
+* The buyer invokes chaincode to record their bid details in their own private data collection.
+  The chaincode is invoked on buyer's peer, and potentially on regulator's peer if required
+  by the collection endorsement policy.
+
+* The current owner (seller) invokes chaincode to sell and transfer the asset, passing in the
+  private details and bid information. The chaincode is invoked on peers of the
+  seller, buyer, and regulator, in order to meet the endorsement policy of the public
+  key, as well as the endorsement policies of the buyer and seller private data collections.
+
+* The chaincode verifies that the submitting client is the owner, verifies the private
+  details against the hash in the seller's collection, and verifies the bid details
+  against the hash in the buyer's collection. The chaincode then writes the proposed
+  updates for the public key (setting ownership to the buyer, and setting endorsement
+  policy to be the buying organization and regulator), writes the private details to the
+  buyer's private data collection, and potentially deletes the private details from seller's
+  collection. Prior to final endorsement, the endorsing peers ensure private data is
+  disseminated to any other authorized peers of the seller and regulator.
+
+* The seller submits the transaction with the public data and private data hashes
+  for ordering, and it is distributed to all channel peers in a block.
+
+* Each peer's block validation logic will consistently verify the endorsement policy
+  was met (buyer, seller, regulator all endorsed), and verify that public and private
+  state that was read in the chaincode has not been modified by any other transaction
+  since chaincode execution.
+
+* All peers commit the transaction as valid since it passed validation checks.
+  Buyer peers and regulator peers retrieve the private data from other authorized
+  peers if they did not receive it at endorsement time, and persist the private
+  data in their private data state database (assuming the private data matched
+  the hashes from the transaction).
+
+* With the transaction completed, the asset has been transferred, and other
+  channel members interested in the asset may query the history of the public
+  key to understand its provenance, but will not have access to any private
+  details unless an owner shares it on a need-to-know basis.
+
+The basic asset transfer scenario could be extended for other considerations,
+for example the transfer chaincode could verify that a payment record is available
+to satisfy payment versus delivery requirements, or verify that a bank has
+submitted a letter of credit, prior to the execution of the transfer chaincode.
+And instead of clients directly hosting peers, they could transact through custodian
+organizations.
```

Comment:
I think we will need some type of enhanced diagram here .... you might also want to think about actually creating a full-fledged example of this ... although that will take some thought.

Reply:
This is actually a teaser and more generic version of the Commercial Paper Advanced sample that Anthony was working on. @ODOWDAIBM do you still intend to complete Commercial Paper Advanced sample in v2.0 timeframe?